api/api.go

```diff
@@ -51,8 +58,17 @@ func Start(elrondFacade MainApiHandler) error {
 	return ws.Run(fmt.Sprintf(":%s", elrondFacade.RestApiPort()))
 }
 
-func joinMonitoringSystem(port string, prometheusJoinURL string) error {
-	req, err := http.NewRequest("POST", prometheusJoinURL, strings.NewReader(port))
+func joinMonitoringSystem(elrondFacade MainApiHandler) error {
+	prometheusJoinUrl := elrondFacade.PrometheusJoinURL()
+	structToSend := prometheusDetails{
+		NodePort:  elrondFacade.RestApiPort(),
+		NetworkID: elrondFacade.PrometheusNetworkID(),
+	}
+	jsonValue, err := json.Marshal(structToSend)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequest("POST", prometheusJoinUrl, bytes.NewBuffer(jsonValue))
```

Comment:
add an empty line before. treat the error.

Reply:
fixed